src/commands/infusion.ts

```diff
@@ -0,0 +1,105 @@
+import { find } from "lodash";
+import { gods } from "../data/gods/all";
+import { AIR, BoonElement, COSMIC, EARTH, FIRE, WATER } from "../data/gods/elements";
+import { InfusionBoon, isInfusion } from "../data/gods/god";
+import { Command } from "./command";
+
+const allElements = [EARTH, WATER, AIR, FIRE, COSMIC];
+const allElementNames = allElements.join("|");
+const infusionCommandExpression = RegExp(`^(infusions?) ?(${allElementNames})?$`, "i");
+
+type InfusionRecord = {
+  boon: string;
+  god: string;
+};
+const elementInfusionMap = gods.reduce<Record<BoonElement, InfusionRecord[]>>(
+  (hash, god) => {
+    // Assumes each god has no more than one infusion, which is currently true.
+    let infusionBoon = find(god.abilities, (boon, _) =>
+      isInfusion(boon)
+    ) as InfusionBoon;
+    if (!infusionBoon || !infusionBoon.requiredElements) {
+      return hash;
+    }
+
+    let record: InfusionRecord = { boon: infusionBoon.name, god: god.name };
+    return {
+      [EARTH]: infusionBoon.requiredElements.find((element) => element === EARTH)
+        ? [...hash[EARTH], record]
+        : hash[EARTH],
+      [WATER]: infusionBoon.requiredElements.find((element) => element === WATER)
+        ? [...hash[WATER], record]
+        : hash[WATER],
+      [AIR]: infusionBoon.requiredElements.find((element) => element === AIR)
+        ? [...hash[AIR], record]
+        : hash[AIR],
+      [FIRE]: infusionBoon.requiredElements.find((element) => element === FIRE)
+        ? [...hash[FIRE], record]
+        : hash[FIRE],
+      [COSMIC]: infusionBoon.requiredElements.find((element) => element === COSMIC)
+        ? [...hash[COSMIC], record]
+        : hash[COSMIC],
+    }
```

Comment:
There is some syntax here that will help you out a lot.
```ts
const requiredElements = infusionBoon.requiredElements
const result = requiredElements.reduce((resultHash, requiredElement) => {
   return { 
       ...resultHash,
      // This syntax lets you define a key where the key value is a variable
       [requiredElement]: [...hash[requiredElement], record]
  }
}, {...hash})
return result;
```

Reply:
Clean! I was following the pattern in commands/elements.ts, but I see the difference now. Actually wonder if we can clean that up similarly.